# Frames
world_frame_id: "world_ned"

# Period
period: 0.05

# Initial state in meters and rads [X, Y, Z, Roll, Pitch, Yaw]
initial_pose: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
initial_velocity: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]

# Vehicle properties
density: 1030.0
mass: 56
buoyancy: 56.3
buoyancy_center: [0.0, 0.0, -0.02]
radius: 0.115
tensor: [0.4, 0.0, 0.0, 0.0, 18.0, 0.0, 0.0, 0.0, 18.0]
damping: [0.0, -4.0, -4.0, -1.0, 0.0, 0.0]
quadratic_damping: [-25.0, -250.0, -250.0, -5.0, -10.8, -10.8]

# Thrusters info (not used if use_force_topic is set to true)
number_of_thrusters: 3
thrusters_tau: 0.1  # Physical property of the thruster
thrusters_max_step: 0.1
thrusters_limiter: [0.6, 1.0, 1.0]
thrusters_symmetric: true
thrusters_matrix: [0.0, 1.0, 1.0,
                   0.0, 0.0, 0.0,
                   1.0, 0.0, 0.0,
                   0.0, 0.0, 0.0,
                   0.0, 0.0, 0.0,
                   0.0, -0.16783, 0.16783]

# Thruster 1
thruster_1_poly_positive: [0.0499, 0.0701, -0.00581, 0.000247, -0.00000467, 0.0000000325]
thruster_1_poly_negative: [0.0220, 0.176, -0.0261, 0.00199, -0.00007026, 0.000000926]
thruster_1_max_force_positive: 55.0
thruster_1_max_force_negative: 28.5

# Thruster 2
thruster_2_poly_positive: [0.06815, 0.05144, -0.00278, 0.0000848, -0.00000118, 0.00000000612]
thruster_2_poly_negative: [0.0772, 0.1186, -0.0139, 0.000920, -0.0000281, 0.000000322]
thruster_2_max_force_positive: 71.5
thruster_2_max_force_negative: 30.0

# Thruster 3
thruster_3_poly_positive: [0.06815, 0.05144, -0.00278, 0.0000848, -0.00000118, 0.00000000612]
thruster_3_poly_negative: [0.0772, 0.1186, -0.0139, 0.000920, -0.0000281, 0.000000322]
thruster_3_max_force_positive: 71.5
thruster_3_max_force_negative: 30.0

# Fins info (not used if fins_topic is set to "")
fins_topic: ""
a_fins: 0.0094
k_cd_fins: 1.8  # Parameters for NACA 0015 airfoil
k_cl_fins: 1.1
max_fins_angle: 0.3941  # 20 degrees = pi / 9 = 0.3491

# If this parameter is set, the input of the model is force
use_force_topic: false
